Authenticator/Source/TokenFormViewController.swift

```diff
@@ -27,14 +27,54 @@ import SVProgressHUD
 
 class TokenFormViewController: UITableViewController {
     private var form: TokenForm?
-    private var viewModel: TableViewModel<Form> = EmptyTableViewModel()
+    private var viewModel: TableViewModel<Form> = EmptyTableViewModel() {
+        didSet {
+            guard oldValue.sections.count == viewModel.sections.count else {
+                // Automatic updates aren't implemented for changing number of sections
+                tableView.reloadData()
+                return
+            }
+            tableView.beginUpdates()
+            for sectionIndex in oldValue.sections.indices {
+                let oldSection = oldValue.sections[sectionIndex]
+                let newSection = viewModel.sections[sectionIndex]
+                let changes = diff(from: ArraySlice(oldSection.rows), to: ArraySlice(newSection.rows), comparator: {
```

Comment:
Length Violation: Line should be 100 characters or less: currently 116 characters